### ISSUE TITLE
one argument indexing not defined for SymTridiagonal, Bidiagonal, Symmetric and Hermitian

### ISSUE THREAD

**Issue by jakebolewski** [JuliaLang/julia#9993]
```julia
julia> r = randn(10,10)
julia>  r = (r' * r) ./ 2
julia> t = SymTridiagonal(r)
julia> t[1]
ERROR: indexing not defined for Base.LinAlg.SymTridiagonal{Float64}
 in getindex at abstractarray.jl:434
```

**Comment by andreasnoack** [comment]
You are having a party tonight!

I'm wondering if we should define a fallback linear indexing method for `AbstractArray`. That would also fix #9991.

**Comment by jakebolewski** [comment]
Updated the title to reflect missing indexing functionality.  Most methods that take` AbstractArrays` as arguments rely on indexing behavior so this is an essential interface for `AbstractArray` subtypes.  Defining a fallback would seem to be the way to go.

**Comment by andreasnoack** [comment]
I think two argument indexing is defined for all the types (if it wasn't they couldn't get printed), so maybe the title should say "One argument indexing...". That would also explain why a fall back would work.

**Comment by mbauman** [comment]
Fixed by #10525.
